cpp/cppsdk/gsdk.cpp

```diff
@@ -289,12 +292,13 @@ namespace Microsoft
                 try {
                     if (heartbeatResponse.isMember("sessionConfig"))
                     {
-                        Json::Value sessionConfig = heartbeatResponse["sessionConfig"];
+						m_configMutex.lock();
```

Comment:
>m_configMutex.lock(); [](start = 6, length = 21)

Usually using a scoped lock is preferred pattern instead of seperate lock/unlock. std::lock_guard works well - please see [https://en.cppreference.com/w/cpp/thread/lock_guard](https://en.cppreference.com/w/cpp/thread/lock_guard) for an example.

Reply:
+1 having a lock that automatically unlocks on going out of scope is goodness, because it deals with cases like exceptions happening.

---
In reply to: [377966260](https://github.com/PlayFab/gsdk/pull/43#discussion_r377966260) [](ancestors = 377966260)